presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1858,7 +1858,7 @@ private static RecursiveDeleteResult doRecursiveDeleteFiles(FileSystem fileSyste
                 boolean eligible = false;
                 // never delete presto dot files
                 if (!fileName.startsWith(".presto")) {
-                    eligible = queryIds.stream().anyMatch(id -> fileName.startsWith(id) || fileName.endsWith(id));
+                    eligible = queryIds.stream().anyMatch(id -> fileName.startsWith(id) || fileName.endsWith(id) || fileName.matches(id + "[.]\\d+$"));
```

Comment:
do you see an option to add test for this?

Reply:
I think this would miss bucketed table files with text format and compression schemes. I created an example table with  `CREATE TABLE hive.default.string_hdfs (val varchar, id integer ) WITH ( external_location = 'hdfs://localhost:9000/tmp/string_hdfs', format = 'TEXTFILE', bucketed_by=ARRAY['val'],bucket_count=2, partitioned_by = ARRAY['id']);` and did an insert with `insert into string_hdfs select 'hello' as a, 1 as b;`
The files created were of the format : `/tmp/string_hdfs/id=1/000000_0_20191118_115616_00022_jrwgx.gz`

Reply:
Does it make sense to change to contains in this case? Or additionally cover all supported formats?

Reply:
I think contains check should work for the current code - but I'm not sure about previous nomenclature.
Probably adding regex to get all extensions (digits or string) should be safer.

Reply:
Also, the regex looks incorrect right now (ref : https://ideone.com/gMcHlG)